asn1tools/compiler.py

```diff
@@ -167,6 +167,42 @@ def decode(self, name, data, check_constraints=False):
 
         return decoded
 
+    def decode_with_length(self, name, data, check_constraints=False):
+        """Decode given bytes object `data` as given type `name` and return
+        the decoded data as a dictionary, along with byte length of the data.
+
+        Use to get length of indefinite-length BER encoded data.
+        Only works for BER or DER codec
+
+        If `check_constraints` is ``True`` all objects in `data` are
+        checked against their ASN.1 type constraints. A
+        ConstraintsError exception is raised if the constraints are
+        not fulfilled. Set `check_constraints` to ``False`` to skip
+        the constraints check and minimize the runtime overhead, but
+        instead allow decoding of values not fulfilling the
+        constraints.
+
+        >>> foo.decode_with_length('Question', b'0\\x0e\\x02\\x01\\x01\\x16\\x09Is 1+1=3?')
+        ({'id': 1, 'question': 'Is 1+1=3?'}, 16)
+
+        """
```

Comment:
This docstring is mostly a copy from the decode function. Add a refernece to the decode function, and wexaplin the differences here.

Reply:
I didn't mean add this commect to the review. Most of the two docstrings are not the same, only the constraints part.